redhat-developer/odo-fork
# Add Reusable Build Container logic for IDP Builds

## What is the purpose of this change? What does it change?

This PR adds in the reusable build container option logic for `kdo build` along with Kube job.

## Was the change discussed in an issue?
{internal}


## How to test changes?

`kdo build <build task type> <project name> <reuse build container flag>`

Reuse the Build Container:
```
./udo build inc projA --reuseBuildContainer
Build arguments: inc projA
reuseBuildContainer flag:  true
Namespace: eclipse-che
Persistent Volume Claim: idp-data-volume
Service Account: default
Checking if Reusable Build Container has already been deployed...
 ✓  Checking to see if a Reusable Container is up... [58ms]
Running pod found: proja-reusable-build-container-7c7959d689-9lspm...
```

Use the Kube Job:
```
./udo build inc projA
Build arguments: inc projA
reuseBuildContainer flag:  false
Namespace: eclipse-che
Persistent Volume Claim: idp-data-volume
Service Account: default
Creating a Kube Job for building...
Creating job codewind-liberty-build-job
Command: [/bin/sh -c] [/data/idp/bin/build-container-update.sh]
The job codewind-liberty-build-job has been created
 ✓  Waiting for the build job to run [4s]
Retrieving job logs for pod: codewind-liberty-build-job-5rqn4
```
